fix(kic): use konghq.com/credentials label instead of kongCredType

czeslavo opened the pull request:
### Description

In KIC 3.0 we deprecated `kongCredType` field in credentials Secrets determining the type of a credential. Instead, `konghq.com/credentials` label should be used.

### Testing instructions

Preview link: <!-- Netlify will generate a preview link after PR is opened. Add links to your edited content here. -->

### Checklist 

- [ ] Review label added <!-- (see below) -->
- [ ] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)


<!-- !!! Only Kong employees can add labels due to a GitHub limitation. If you're an OSS contributor, thank you! The maintainers will label this PR for you !!! -->

<!-- When raising a pull request, indicate what type of review you need with one of the following labels:

    review:copyedit: Request for writer review.
    review:general: Review for general accuracy and presentation. Does the doc work? Does it output correctly?
    review:tech: Request for technical review for a docs platform change.
    review:sme: Request for review from an SME (engineer, PM, etc).

At least one of these labels must be applied to a PR or the build will fail.
-->

rainest commented on app/_includes/plugins/kubernetes_consumer.html:
```diff
@@ -4,8 +4,11 @@
+  {% for field in include.hub_example.auth_fields -%}--from-literal={{ field[0] }}={{ field[1] }} \
+  {% endfor -%}
+  -o json --dry-run=client | \
+  jq '.metadata.labels |= {"konghq.com/credentials": "{{ include.hub_example.plugin_name }}" }' | \
```
```suggestion
  jq '.metadata.labels |= {"konghq.com/credential": "{{ include.hub_example.plugin_name }}" }' | \
```

rainest commented on app/_includes/md/kic/key-auth.md:
```diff
@@ -3,8 +3,10 @@
-      --from-literal=key={{ key }}
+      --from-literal=key={{ key }} \
+      -o json --dry-run=client | \
+      jq '.metadata.labels |= {"konghq.com/credentials": "key-auth"}' | \
```
```suggestion
      jq '.metadata.labels |= {"konghq.com/credential": "key-auth"}' | \
```

rainest commented on app/_src/kic-v2/deployment/admission-webhook.md:
```diff
@@ -246,8 +246,10 @@ for incorrect secret types and wrong parameters to the secrets.
-  --from-literal=username=foo
+  --from-literal=username=foo \
+  -o json --dry-run=client | \
+  jq '.metadata.labels |= {"konghq.com/credentials": "basic-auth"}' | \
```
```suggestion
  jq '.metadata.labels |= {"konghq.com/credential": "basic-auth"}' | \
```

rainest commented on app/_src/kic-v2/deployment/admission-webhook.md:
```diff
@@ -256,8 +258,10 @@ Error from server: admission webhook "validations.kong.konghq.com" denied the re
-  --from-literal=sdfkey=my-sooper-secret-key
+  --from-literal=sdfkey=my-sooper-secret-key \
+  -o json --dry-run=client | \
+  jq '.metadata.labels |= {"konghq.com/credentials": "wrong-auth"}' | \
```
```suggestion
  jq '.metadata.labels |= {"konghq.com/credential": "wrong-auth"}' | \
```

rainest commented on app/_src/kic-v2/guides/allowing-multiple-authentication-methods.md:
```diff
@@ -67,9 +67,11 @@ Create two consumers that use different authentication methods:
-      --from-literal=password=consumer-1-password
+      --from-literal=password=consumer-1-password \
+      -o json --dry-run=client | \
+      jq '.metadata.labels |= {"konghq.com/credentials": "basic-auth"}' | \
```
```suggestion
      jq '.metadata.labels |= {"konghq.com/credential": "basic-auth"}' | \
```

rainest commented on app/_src/kic-v2/guides/configure-acl-plugin.md:
```diff
@@ -108,17 +108,21 @@ C6V0e/O3LEuJrP+XrEndtLsCAwEAAQ==
-      --from-literal=rsa_public_key="{{ public_key }}"
+      --from-literal=rsa_public_key="{{ public_key }}" \
+      -o json --dry-run=client | \
+      jq '.metadata.labels |= {"konghq.com/credentials": "jwt"}' | \
```
```suggestion
      jq '.metadata.labels |= {"konghq.com/credential": "jwt"}' | \
```

rainest commented on app/_src/kic-v2/guides/configure-acl-plugin.md:
```diff
@@ -108,17 +108,21 @@ C6V0e/O3LEuJrP+XrEndtLsCAwEAAQ==
-      --from-literal=rsa_public_key="{{ public_key }}"
+      --from-literal=rsa_public_key="{{ public_key }}" \
+      -o json --dry-run=client | \
+      jq '.metadata.labels |= {"konghq.com/credentials": "jwt"}' | \
```
```suggestion
      jq '.metadata.labels |= {"konghq.com/credential": "jwt"}' | \
```

rainest commented on app/_src/kic-v2/guides/configure-acl-plugin.md:
```diff
@@ -281,13 +285,17 @@ httproute.gateway.networking.k8s.io/lime annotated
-      --from-literal=group=admin
+      --from-literal=group=admin \
+      -o json --dry-run=client | \
+      jq '.metadata.labels |= {"konghq.com/credentials": "acl"}' | \
```
```suggestion
      jq '.metadata.labels |= {"konghq.com/credential": "acl"}' | \
```

rainest commented on app/_src/kic-v2/guides/configure-acl-plugin.md:
```diff
@@ -281,13 +285,17 @@ httproute.gateway.networking.k8s.io/lime annotated
-      --from-literal=group=user
+      --from-literal=group=user \
+      -o json --dry-run=client | \
+      jq '.metadata.labels |= {"konghq.com/credentials": "acl"}' | \
```
```suggestion
      jq '.metadata.labels |= {"konghq.com/credential": "acl"}' | \
```

rainest reviewed (changes requested):
Plural/not plural confusion remains an evergreen source of fun: https://github.com/Kong/kubernetes-ingress-controller/blob/v3.0.1/internal/labels/labels.go#L10

Unfortunately, it looks like a `create secret` or whatever `kubectl` helper is out of the question per https://github.com/kubernetes/kubernetes/issues/60295 (the `--override` suggestion is a lie, that only works with `kubectl run`).

IMO we should avoid the `jq` dependency by using a full manifest, i.e. we can use

```
echo '
apiVersion: v1
kind: Secret
metadata:
  annotations:
    konghq.com/credential: whatever
stringData:
  key: "i-am-a-key!"
' | kubectl apply -f -
```
similar to most other resources that lack a shorthand CLI. Keeping the shorthand would be nice, but if it's copy-pastable still, not the worst loss.

Rajakavitha1 reviewed (changes requested):
update the 3.0 guides as well.

mheap commented:
@czeslavo Did you mean to update the KIC 2.x docs with this change?

czeslavo commented:
> @czeslavo Did you mean to update the KIC 2.x docs with this change?

No, I didn't. My mistake.

**Update**: Converted all `kubectl create secret` with `kongCredType` to `echo {manifest} | kubectl apply -f -` (didn't manage to do that in `app/_src/kubernetes-ingress-controller/plugins/acl.md` because of a multiline public key string that I couldn't make to indent properly in the rendered yaml, backed off to a separate `kubectl label secrets` call which should be fine).

Fixed pages:
- https://deploy-preview-6556--kongdocs.netlify.app/kubernetes-ingress-controller/latest/concepts/admission-webhook/ 
- https://deploy-preview-6556--kongdocs.netlify.app/hub/kong-inc/basic-auth/how-to/basic-example/ (and other plugins using `kubernetes_consumer.html` template)
- https://deploy-preview-6556--kongdocs.netlify.app/kubernetes-ingress-controller/latest/plugins/acl/
- https://deploy-preview-6556--kongdocs.netlify.app/kubernetes-ingress-controller/latest/plugins/authentication/#create-consumers-and-secrets

mloskot commented:
Shouldn't this the example at https://docs.konghq.com/kubernetes-ingress-controller/3.1.x/concepts/ingress be also ported from `kongCredType` to the label?

https://github.com/Kong/docs.konghq.com/blob/b4fcbc6af7d9b27f8d672938566f699dfd9bd9f3/app/_src/kubernetes-ingress-controller/concepts/ingress.md?plain=1#L75-L81

Thanks for these updates to the docs @czeslavo !
